### PR TITLE
feature: human readable cloud provider

### PR DESCRIPTION
**Pull request by shashvatshah9**
## Reference Issues or PRs

Issue targetted [issue-1708](https://github.com/nebari-dev/nebari/issues/1708)
Updated the ProviderEnum values.
Changing the value raised some other issues, so updated the `enum_to_list()` to add a  case to avoid to lower() a string.
Also updated a few lines where comparison was made with Enum values

## What does this implement/fix?

- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [x ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ x] Did you test the pull request locally?
- [ ] Did you add new tests?

## Any other comments?

Some tests are failing on my local, and it needs to be checked if there is some value hardcoding in tests. Need help on this.

### CODE REVIEW

**Review by iameskild — APPROVED** [review]
Thank you for the contribution @shashvatshah9 :)

**Comment by dharhas** [comment]
@shashvatshah9 @iameskild 

I think we should make the following changes before we merge:

1. We are not using `lower=False` so we should remove that change
2. I have some suggestions on the new text.

Instead of 

>   local = "local"
>   existing = "existing K8s cluster"
>   do = "DigitalOcean"
>   aws = "Amazon Web Services (AWS)"
>   gcp = "Google Cloud Platform (GCP)"
>   azure = "Microsoft Azure"

Lets clean up slightly for consistency

>  local = "Local test k8s cluster (using 'kind')"
>  existing = "Existing k8s cluster"
>  do = "DigitalOcean (DO)"
>  aws = "Amazon Web Services (AWS)"
>  gcp = "Google Cloud Platform (GCP)"
>  azure = "Microsoft Azure (Azure)"

**Comment by iameskild** [comment]
Hey @shashvatshah9, this PR looks like it's really close to being done. Do you have time to wrap this up? Thanks :)

**Comment by shashvatshah9** [comment]
> Hey @shashvatshah9, this PR looks like it's really close to being done. Do you have time to wrap this up? Thanks :)

Updated the ProviderEnum with the latest string values, but tests are failing due to missing env variable in local NEBARI_DISABLE_INIT_CHECKS. So need some help here with test configuration.

FAILED tests/test_dependencies.py::test_build_by_conda_forge - subprocess.CalledProcessError: Command '['conda', 'build', PosixPath('/private/tmp/pytest-of-root/pytest-0/te...
FAILED tests/test_schema.py::test_schema[setup_fixture0] - KeyError: 'provider'
FAILED tests/test_schema.py::test_schema[setup_fixture1] - KeyError: 'provider'
FAILED tests/test_schema.py::test_schema[setup_fixture2] - KeyError: 'provider'
FAILED tests/test_schema.py::test_schema[setup_fixture3] - KeyError: 'provider'
FAILED tests/test_upgrade.py::test_upgrade_4_0[./qhub-config-yaml-files-for-upgrade/qhub-config-do-310.yaml-False-False] - KeyError: 'provider'
FAILED tests/test_upgrade.py::test_upgrade_4_0[./qhub-config-yaml-files-for-upgrade/qhub-config-do-310-customauth.yaml-False-True] - KeyError: 'provider'
FAILED tests/test_upgrade.py::test_upgrade_4_0[./qhub-config-yaml-files-for-upgrade/qhub-config-do-310-customauth.yaml-True-False] - KeyError: 'provider'

**Comment by shashvatshah9** [comment]
Also, we do need lower=False in enum_to_list() method call. It not an ideal way of doing things, but somehow that's how the comparisons are being done with enums in this codebase

**Review by iameskild — CHANGES_REQUESTED** [review]
Hi @shashvatshah9 sorry for all of the back and forth but I commented below that I think we ought to keep the `ProviderEnum` unmodified; I provided a potential alternative. 

Thanks again for your contribution! We appreciate you taking the time to work with us :)

File: nebari/schema.py [hunk 1]
```diff
@@ -24,12 +24,12 @@ class TerraformStateEnum(str, enum.Enum):
...
-    do = "do"
-    aws = "aws"
-    gcp = "gcp"
-    azure = "azure"
+    local = "Local test k8s cluster (using 'kind')"
+    existing = "Existing k8s cluster"
+    do = "DigitalOcean (DO)"
+    aws = "Amazon Web Services (AWS)"
+    gcp = "Google Cloud Platform (GCP)"
+    azure = "Microsoft Azure (Azure)"
```
[the remark this answers is not in the document]
Comment (pavithraes) [reply nebari/schema.py:32]: @shashvatshah9 Hi and thanks for contributing this PR! I'm pinging to surface this comment, would you have the time to incorporate `@iameskild`'s suggestion? It'll be great to get this PR merged. :)

[the remark this answers is not in the document]
Comment (iameskild) [reply nebari/schema.py:32]: Hi @shashvatshah9, if it's alright with you, I will be wrapping up this PR later this week.

[the remark this answers is not in the document]
Comment (iameskild) [reply nebari/schema.py:32]: Hi @shashvatshah9, if it's alright with you, I will be wrapping up this PR later this week.

**Comment by iameskild** [comment]
Hi @shashvatshah9, we will be closing this PR for now. There is a large refactor in the work that will incorporate the changes you proposed so thank you for your contributions :)

**Pull request closed without merging**